### PR TITLE
Add admin user customization kill switch

### LINKED ISSUE #34561: Theming: An admin should be able to disable account-specific customization completely
In the case of white labeling, an admin might not want their users to be able to change a branded background or to pick colors that are not compatible with it.

For this case, an admin should be able to disable this account-specific customization completely, if they would like to keep their cloud on-brand. This would be a setting in "Theming" called "Allow people to personalize their background and color". By default, the customization is enabled.

### PR DESCRIPTION
**Pull request by skjnldsv**
(ignore the missing icons sometimes, the chromium screenshot tool bugged)

|| Customization enabled | Disabled |
|:---------:|:------:|:------:|
|Default|default-user-enabled|default-user-disabled|
|Custom bg|custom-user-enabled|custom-bg-user-disabled|
|No bg|-|no-bg-user-disabled|

Fix https://github.com/nextcloud/server/issues/34561

image
image

### CODE REVIEW

**Comment by szaimen** [comment]
@skjnldsv is this ready for review?

**Comment by skjnldsv** [comment]
> @skjnldsv is this ready for review?

Yes

**Review by szaimen — APPROVED** [review]
works great and fixes all the problems that I found a few days ago. (but didnt review the code)

The only problem that I found is this one (it shows the primary background when changing user theming all over the screen which is really disturbing) but not a blocker imho:

https://user-images.githubusercontent.com/42591237/196505135-9d5af980-447f-4403-8c7c-72435dd554c5.mp4

**Review by Pytal — APPROVED** [review]

File: apps/theming/lib/Themes/DefaultTheme.php [hunk 1]
```diff
@@ -237,10 +238,11 @@ public function getCSSVariables(): array {
 
 		$appManager = Server::get(IAppManager::class);
 		$user = $this->userSession->getUser();
-		if ($appManager->isEnabledForUser(Application::APP_ID) && $user !== null) {
+		if (!$this->themingDefaults->isUserThemingDisabled() && $appManager->isEnabledForUser(Application::APP_ID) && $user !== null) {
 			$themingBackground = $this->config->getUserValue($user->getUID(), Application::APP_ID, 'background', 'default');
 			$currentVersion = (int)$this->config->getUserValue($user->getUID(), Application::APP_ID, 'userCacheBuster', '0');
 
+
```
Comment (Pytal) [inline apps/theming/lib/Themes/DefaultTheme.php:245]:
```suggestion
```

**Pull request merged**